libchisel/src/deployer.rs

```diff
@@ -0,0 +1,190 @@
+use parity_wasm::builder;
+
+use super::*;
+
+use byteorder::{LittleEndian, WriteBytesExt};
+use rustc_hex::FromHex;
+
+fn deployer_code() -> Vec<u8> {
+    FromHex::from_hex(
+        "
+        0061736d010000000113046000017f60037f7f7f0060027f7f00600000023e0308
+        657468657265756d0b676574436f646553697a65000008657468657265756d0863
+        6f6465436f7079000108657468657265756d0666696e6973680002030201030503
+        010001071102066d656d6f72790200046d61696e00030a2c012a01037f10002100
+        4100410020001001200041046b2802002102200041046b20026b21012001200210
+        020b
+    ",
+    ).unwrap()
+}
+
+/// Returns a module which contains the deployable bytecode as a custom section.
+pub fn create_custom_deployer(payload: &[u8]) -> Module {
+    // The standard deployer code, which expects a 32 bit little endian as the trailing content
+    // immediately following the payload, placed in a custom section.
+    let code = FromHex::from_hex(
+        "
+        0061736d010000000113046000017f60037f7f7f0060027f7f00600000023e0308
+        657468657265756d0b676574436f646553697a65000008657468657265756d0863
+        6f6465436f7079000108657468657265756d0666696e6973680002030201030503
+        010001071102066d656d6f72790200046d61696e00030a2c012a01037f10002100
+        4100410020001001200041046b2802002102200041046b20026b21012001200210
+        020b
+    ",
+    ).unwrap();
+    // This is the pre-written deployer code.
+    let mut module: Module = parity_wasm::deserialize_buffer(&code).expect("Failed to load module");
+
+    // Prepare payload (append length).
+    let mut custom_payload = payload.to_vec();
+    custom_payload
+        .write_i32::<LittleEndian>(payload.len() as i32)
+        .unwrap();
+
+    // Prepare and append custom section.
+    let custom = CustomSection {
+        name: "deployer".to_string(),
```

Comment:
Instead of having to fork an old version of parity, can we stick to the 0.21 tag and just implement a trait over `CustomSection` with a proper constructor? It would be much easier to maintain.

Reply:
I'd rather do it outside of this PR.

Reply:
If you implement a trait for a type, the trait has access to its type's members via `&self`

Reply:
Ah, so a trait _after_ the instance has been created.

Reply:
Not quite sure what you meant by that.

Reply:
Discussed this offline.